finality-aleph/src/new_network/component.rs

```diff
@@ -36,6 +36,59 @@ impl<D: Data, CN: Network<D>> DataNetwork<D> for CN {
 #[async_trait::async_trait]
 impl<D: Data> Receiver<D> for mpsc::UnboundedReceiver<D> {
     async fn next(&mut self) -> Option<D> {
-        self.next().await
+        StreamExt::next(self).await
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{Network, Receiver, Sender};
+    use crate::new_network::{SendError};
+    use aleph_bft::Recipient;
+    use futures::channel::mpsc;
+    use std::sync::Arc;
+    use tokio::sync::Mutex;
+    use std::ops::DerefMut;
+
+    impl Sender<u64> for mpsc::UnboundedSender<u64> {
+        fn send(&self, data: u64, _recipient: Recipient) -> Result<(), SendError> {
+            self.unbounded_send(data).map_err(|_| SendError::SendFailed)
+        }
+    }
+
+    struct TestNetwork {
+        sender: mpsc::UnboundedSender<u64>,
+        receiver: Arc<Mutex<mpsc::UnboundedReceiver<u64>>>,
+    }
+
+    impl Network<u64> for TestNetwork {
+        type S = mpsc::UnboundedSender<u64>;
+
+        type R = mpsc::UnboundedReceiver<u64>;
+        
+        fn sender(&self) -> &Self::S {
+            &self.sender
+        }
+        
+        fn receiver(&self) -> Arc<Mutex<Self::R>> {
+            self.receiver.clone()
+        }
+    }
```

Comment:
Everything above this line in the test is IMO unnecessary – this is not how you would implement a `Network` anyway, and even if it was it doesn't really test anything. Just create a receiver channel and test it by calling `Receiver::next(receiver)` on it. You don't even have to implement sender for the other end of the channel in this case.

This is a nit obviously, but leaving this in might result in confusion as to what the test is actually about.

Reply:
Done, but due to `cargo fmt `now we are proposing the same changes :sweat_smile: